### PR TITLE
[DeepseekV32]: use `_concat_mla_absorb_q_general` to replace `torch.cat`

### PR DESCRIPTION
**Pull request by bingps**
## Motivation

https://github.com/sgl-project/sglang/issues/11989 
`torch.cat([q_nope, q_rope], dim=-1)` is heavily used in `nsa_backend`, which is less efficient.

## Modifications

Use existed kernel `_concat_mla_absorb_q_general` to replace.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

**Comment by Fridge003** [comment]
@bingps Can you please post some accuracy results, as well as performance benchmarks before/after this change?

**Comment by bingps** [comment]
> @bingps Can you please post some accuracy results, as well as performance benchmarks before/after this change?

Sure~ Here is a simple benchmark.

```python
import torch
import triton
from sglang.srt.layers.attention.trtllm_mla_backend import _concat_mla_absorb_q_general


if __name__ == "__main__":
    H = 128
    D_NOPE = 512
    D_ROPE = 64

    for S in [1, 128, 256, 512, 1024, 2048, 4096, 8192, 16384, 30000]:
        print(f"============ Test _concat_mla_absorb_q_general on {S=} ============")
        q_nope = torch.randn(S, H, D_NOPE, dtype=torch.bfloat16, device="cuda")
        q_rope = torch.randn(S, H, D_ROPE, dtype=torch.bfloat16, device="cuda")

        q_ref = torch.cat([q_nope, q_rope], dim=-1)
        q = _concat_mla_absorb_q_general(q_nope, q_rope)
        assert torch.equal(q_ref, q)

        quantiles = [0.5, 0.2, 0.8]
        ms, min_ms, max_ms = triton.testing.do_bench(
            lambda: torch.cat([q_nope, q_rope], dim=-1),
            quantiles=quantiles,
        )
        print(f"Torch  cat: median {ms*1000:4.0f} us, min {min_ms*1000:4.0f} us, max {max_ms*1000:4.0f} us")

        ms, min_ms, max_ms = triton.testing.do_bench(
            lambda: _concat_mla_absorb_q_general(q_nope, q_rope),
            quantiles=quantiles,
        )
        print(f"concat_mla: median {ms*1000:4.0f} us, min {min_ms*1000:4.0f} us, max {max_ms*1000:4.0f} us")
```

The concat results are bit-wise equal, and it is 2.4x faster for 16K concat, while not slower when S=1.
```
============ Test _concat_mla_absorb_q_general on S=1 ============
Torch  cat: median    7 us, min    7 us, max    7 us
concat_mla: median    6 us, min    6 us, max    7 us
============ Test _concat_mla_absorb_q_general on S=128 ============
Torch  cat: median   32 us, min   32 us, max   32 us
concat_mla: median   18 us, min   17 us, max   18 us
============ Test _concat_mla_absorb_q_general on S=256 ============
Torch  cat: median   57 us, min   57 us, max   57 us
concat_mla: median   28 us, min   28 us, max   29 us
============ Test _concat_mla_absorb_q_general on S=512 ============
Torch  cat: median  106 us, min  106 us, max  107 us
concat_mla: median   49 us, min   48 us, max   50 us
============ Test _concat_mla_absorb_q_general on S=1024 ============
Torch  cat: median  206 us, min  206 us, max  206 us
concat_mla: median   90 us, min   90 us, max   91 us
============ Test _concat_mla_absorb_q_general on S=2048 ============
Torch  cat: median  405 us, min  405 us, max  406 us
concat_mla: median  174 us, min  172 us, max  174 us
============ Test _concat_mla_absorb_q_general on S=4096 ============
Torch  cat: median  804 us, min  803 us, max  804 us
concat_mla: median  340 us, min  340 us, max  341 us
============ Test _concat_mla_absorb_q_general on S=8192 ============
Torch  cat: median 1599 us, min 1599 us, max 1600 us
concat_mla: median  672 us, min  671 us, max  673 us
============ Test _concat_mla_absorb_q_general on S=16384 ============
Torch  cat: median 3190 us, min 3183 us, max 3191 us
concat_mla: median 1337 us, min 1337 us, max 1338 us
```

**Comment by bingps** [comment]
However in my testing, the kernel fails for ~30K inputs :crying_cat_face:

See https://github.com/sgl-project/sglang/issues/12250

**Comment by Johnsonms** [comment]
> However in my testing, the kernel fails for ~30K inputs 😿
> 
> See #12250

me too

**Comment by Fridge003** [comment]
Might wait for another sgl-kernel bump to include #12453

**Review by Fridge003 — APPROVED** [review]
Waiting for CI
